File: Design-Patterns/Mediator/Cpp/Mediator/lazy_programmer.cpp
#include <ctime>
#include "employee.h"

lazy_programmer::lazy_programmer(std::string name, manager* manager) : programmer(name, manager)
{

}

void lazy_programmer::do_job()
{
	if (jobs_.empty()) return;
	if (mood_ == angry || mood_ == bad || mood_ == normal || mood_ == good) return;

	srand(time(nullptr));
	int index = rand() % jobs_.size();
	job completed_job = jobs_[index];
	jobs_.erase(jobs_.begin() + index);
	manager_->transfer_job(completed_job, this);
}

File: Design-Patterns/Mediator/CSharp/Mediator/Employees/LazyProgrammer.cs
using System;
using Mediator.Managers;

namespace Mediator.Employees
{
    public class LazyProgrammer : Programmer
    {
        public LazyProgrammer(string name, Manager manager) : base(name, manager)
        {

        }

        public override void DoJob()
        {
            if (Jobs.Count == 0) return;
            if (Mood == Mood.Angry || Mood == Mood.Bad || Mood == Mood.Normal || Mood == Mood.Good) return;

            int index = new Random().Next(Jobs.Count);
            Job completedJob = Jobs[index];
            Jobs.RemoveAt(index);
            Manager.TransferJob(completedJob, this);
        }
    }
}
